packages/js/src/ui/components/primitives/Switch.tsx

```diff
@@ -1,30 +1,71 @@
-import { useStyle } from '../../helpers';
+import { cn, useStyle } from '../../helpers';
 
-type SwitchProps = {
-  checked?: boolean;
-  onChange: (checked: boolean) => void;
+export type SwitchState = 'enabled' | 'disabled' | 'indeterminate';
+
+export type SwitchProps = {
+  state?: SwitchState;
+  onChange: (state: SwitchState) => void;
+  disabled?: boolean;
 };
 
 export const Switch = (props: SwitchProps) => {
   const style = useStyle();
 
+  const handleChange = () => {
+    if (props.disabled) return;
+
+    const nextState = getNextState(props.state ?? 'disabled');
+    props.onChange(nextState);
+  };
+
+  const getNextState = (currentState: SwitchState): SwitchState => {
```

Comment:
switch component changes to support three states: enabled, disabled, indeterminate